### PR TITLE
Migration Instructions/Doc: cosmoshub-2

### PR DESCRIPTION
**Pull request by alexanderbez**
Moving the manual migration instructions from the wiki to a doc in the repo to facilitate collaboration with the community.

I've already made some tweaks and prettified the markdown. Looking for community feedback.

----

- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: docs/migration/cosmoshub-2.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+3. Verify the SHA256 of the (sorted) exported genesis file:
+
+```shell
+$ jq -S -c -M '' cosmoshub_2_genesis_export.json | shasum -a 256
+[PLACEHOLDER]  cosmoshub_2_genesis_export.json
+```
+
+4. At this point you now have a valid exported genesis state! All further steps now require
+v2.0.3 of [Gaia](https://github.com/cosmos/gaia).
+
```
Comment (dimiandre) [inline docs/migration/cosmoshub-2.md:126]: We should add a note about golang version, check that is 1.13.x or upgrade

Comment (alexanderbez) [reply docs/migration/cosmoshub-2.md:126]: updated.

**Comment by 1xCL** [comment]
Before `$ gaiad export --for-zero-height --height=[PLACEHOLDER] > cosmoshub_2_genesis_export.json`, I think it should mention about stopping the gaiad node.

Before `git checkout v2.0.3; make install`, since `v2.0.3` is on `gaia` repository and `v0.34.6+` is on `cosmos-sdk` repository, it would be good to mention `git clone https://github.com/cosmos/gaia.git` first. And for `All further steps now require v2.0.3 of Gaia and Go 1.13+.`, I think it would be a good idea to put more emphasis on upgrading Go to `1.13+`.

And again perhaps mention about the backup one more time before `$ gaiad unsafe-reset-all`.

**Comment by alexanderbez** [comment]
updated @winslyn

**Review by jackzampolin — APPROVED** [review]

**Review by fedekunze — APPROVED** [review]

**Pull request merged**